### PR TITLE
Basic Anti-Bot Protection

### PR DESCRIPTION
**Pull request by Gingerbreadfork**
1.) Simple colored shapes have been used on buttons that shuffle and require a new user to select the "green circle" this allows those that don't speak English or happen to be color blind to complete the entry test. If a user fails they are simply restricted from posting which should reduce the issues with accidental failures as they can contact an admin they see in the chat.

2.) Welcome message has also been adjusted to avoid rollover on mobile.

3.) Was unable to find a way to time delete the messages, but showing the link preview on the message has been disabled to avoid it being too large (suggestions on how to remove the messages after a set amount of time would be great if you have any ideas I hit a wall trying to get this to work).

image

### CODE REVIEW

File: localcoinbot.py [hunk 1]
```diff
@@ -199,7 +244,7 @@ def debug_group_id(update, context):
 
 def main():
     # Start the bot
-    updater = Updater(token=TOKEN)
+    updater = Updater(str(TOKEN), use_context=True)
```
Comment (CodeForcer) [inline localcoinbot.py]: @Gingerbreadfork was this due to the update of the package? Generally we try and avoid too much casting in Python for readability. Only a miner issue though

File: localcoinbot.py [hunk 2]
```diff
@@ -82,23 +79,75 @@ def admins(update, context):
...
+                keyboard.append([])
+                for n in range(3):
+                    keyboard_item = keyboard_items[counter]
+                    keyboard[i].append(keyboard_item)  # fills nested lists with data
+                    counter += 1
+
+            reply_markup = InlineKeyboardMarkup(keyboard)
+
+
+            context.bot.sendMessage(
```
Comment (CodeForcer) [inline localcoinbot.py]: @Gingerbreadfork what does this return? Maybe the message ID?

**Review by CodeForcer — CHANGES_REQUESTED** [review]
I think this is great, only thing missing is the message delete functionality. I think we need that or we are going to be annoying people in the chat.

It will be great to drop that annoying welcome message that plays every 3 joiners

File: localcoinbot.py (hunk 1 above)
Comment (Gingerbreadfork) [reply localcoinbot.py]: use_context param here is required for v12 of the python-telegram-bot package but won't be required for v13, just some of the magic of this package apparently.

**Review by CodeForcer — APPROVED** [review]
Happy to merge, looks good

**Pull request merged**